src/applications/check-in/travel-claim/pages/travel-mileage/MultiFacilityBody.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { VaCheckbox } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { useTranslation } from 'react-i18next';
+
+const MultipleFacilityBody = props => {
+  const {
+    error,
+    formatAppointment,
+    appointmentsByFacility,
+    selectedFacilities,
+    setSelectedFacilities,
+  } = props;
+
+  const { t } = useTranslation();
+
+  const onCheck = useCallback(
+    e => {
+      const stationNo = e.target.value;
+      const { checked } = e.detail;
+      const appointments = appointmentsByFacility[stationNo];
+      const firstAppointment = appointments[0];
+      const value = {
+        stationNo: firstAppointment.stationNo,
+        startTime: firstAppointment.startTime,
+        multipleAppointments: appointments.length > 1,
+      };
+      let newFacilities;
```

Comment:
If this was it's own function that returned new facilities you could test the logic easier.